### PR TITLE
Some SwiftUI fixes and improvements

### PR DESCRIPTION
**Pull request by janst97**
- fix list style in OMEMO key view
- SwiftUI QR Code Scanner functionality moved to a single class (no more
  ScannerViews)
- Information alert that explains own device key
- Fix crash when using SwiftUI preview on ContactDetails

### CODE REVIEW

File: Monal/Classes/MLQRCodeScanner.swift [hunk 1]
```diff
@@ -305,3 +306,67 @@ struct XMPPLoginQRCode : Codable
...
+            self.handleClose()
+        }
+
+        init(handleLogin: ((String, String) -> Void)?, handleContact: ((String, Dictionary<NSInteger, String>) -> Void)?, handleClose: @escaping () -> Void) {
+            self.handleLogin = handleLogin
+            self.handleContact = handleContact
+            self.handleClose = handleClose
+        }
+
+        func MLQRCodeContactScanned(jid: String, fingerprints: Dictionary<NSInteger, String>) {
```
Comment (FriedrichAltheide) [inline Monal/Classes/MLQRCodeScanner.swift]: move above closeQRCodeScanner

File: Monal/Classes/OmemoKeys.swift [hunk 2]
```diff
@@ -245,15 +257,15 @@ struct OmemoKeys: View {
...
-        let knownDevices = Array(account.omemo.knownDevices(forAddressName: self.scannedJid))
-        for (qrDeviceId, fingerprint) in self.scannedFingerprints {
+    func resetTrustFromQR(scannedJid : String, scannedFingerprints : Dictionary<NSInteger, String>) {
+        let knownDevices = Array(self.account!.omemo.knownDevices(forAddressName: scannedJid))
+        for (qrDeviceId, fingerprint) in scannedFingerprints {
             if(knownDevices.contains(NSNumber(integerLiteral: qrDeviceId))) {
-                let address = SignalAddress(name: self.scannedJid, deviceId: Int32(qrDeviceId))
-                let identity = account.omemo.getIdentityFor(address)
+                let address = SignalAddress(name: scannedJid, deviceId: Int32(qrDeviceId))
+                let identity = account!.omemo.getIdentityFor(address)
```
Comment (FriedrichAltheide) [inline Monal/Classes/OmemoKeys.swift]: self?

File: Monal/Classes/OmemoKeys.swift [hunk 3]
```diff
@@ -263,7 +275,20 @@ struct OmemoKeys: View {
...
+                    for contact in self.contacts {
+                        if(contact.obj.contactJid == self.scannedJid) {
+                            resetTrustFromQR(scannedJid: jid, scannedFingerprints: fingerprints)
+                            return
+                        }
+                    }
+                    // we scanned a contact but it was not in the contact list, show the alert...
+                    self.scannedJid = jid
+                    self.scannedFingerprints = fingerprints
+                    showScannedContactMissmatchAlert = true
```
Comment (FriedrichAltheide) [inline Monal/Classes/OmemoKeys.swift:283]: Please ask if the user wants to add the contact to account...

Comment (janst97) [reply Monal/Classes/OmemoKeys.swift:283]: That's exactly what the code does (unless you want a prompt even if the user is in the contact list of the group).

**Pull request merged**